modules/objdetect/include/opencv2/objdetect/checker_detector.hpp

```diff
@@ -0,0 +1,222 @@
+ * - double findCandidatesApproxPolyDPEpsMultiplier : multipler to be used in
+ *                                                    cv::ApproxPolyDP function
+ *                                                    (default 0.05)
+ * - int borderWidth : width of the padding used to pass the inital neural
+ *                     network detection in the succeeding system.(default 0)
+ * - float B0factor : distance between two neighbours squares of the same chart.
+ *                    Defined as the ratio between distance and large dimension
+ *                    of square (default 1.25)
+ * - float maxError : maximum allowed error in the detection of a chart.
+ *                    default(0.1)
+ * - int minContourPointsAllowed : minium points in a detected contour.
+ *                                 default(4)
+ * - int minContourLengthAllowed : minimum length of a countour. default(100)
+ * - int minInterContourDistance : minimum distance between two contours.
+ *                                 default(100)
+ * - int minInterCheckerDistance : minimum distance between two checkers.
+ *                                 default(10000)
+ * - int minImageSize : minimum size of the smaller dimension of the image.
+ *                      default(1000)
+ * - unsigned minGroupSize : minimum number of a squared of a chart that must be
+ *                           detected. default(4)
+ */
+struct CV_EXPORTS_W DetectorParametersMCC
+{
+
+    DetectorParametersMCC();
+
+    CV_WRAP static Ptr<DetectorParametersMCC> create();
+
+    CV_PROP_RW int adaptiveThreshWinSizeMin;
+    CV_PROP_RW int adaptiveThreshWinSizeMax;
+    CV_PROP_RW int adaptiveThreshWinSizeStep;
+    CV_PROP_RW double adaptiveThreshConstant;
+    CV_PROP_RW double minContoursAreaRate;
+    CV_PROP_RW double minContoursArea;
+    CV_PROP_RW double confidenceThreshold;
+    CV_PROP_RW double minContourSolidity;
+    CV_PROP_RW double findCandidatesApproxPolyDPEpsMultiplier;
+    CV_PROP_RW int borderWidth;
+    CV_PROP_RW float B0factor;
+    CV_PROP_RW float maxError;
+    CV_PROP_RW int minContourPointsAllowed;
+    CV_PROP_RW int minContourLengthAllowed;
+    CV_PROP_RW int minInterContourDistance;
+    CV_PROP_RW int minInterCheckerDistance;
+    CV_PROP_RW int minImageSize;
+    CV_PROP_RW unsigned minGroupSize;
+};
+
+/** @brief A class to find the positions of the ColorCharts in the image.
+ */
+
+class CV_EXPORTS_W CCheckerDetector : public Algorithm
+{
+public:
+    /** \brief Set the net which will be used to find the approximate
+    *         bounding boxes for the color charts.
+    *
+    * It is not necessary to use this, but this usually results in
+    * better detection rate.
+    *
+    * \param net the neural network, if the network in empty, then
+    *            the function will return false.
+    * \return true if it was able to set the detector's network,
+    *         false otherwise.
+    */
+
+    CV_WRAP virtual bool setNet(dnn::Net net) = 0;
```

Comment:
Let it be a `create()` factory parameter like in https://github.com/opencv/opencv/pull/26875